src/styles/components/_calendar.scss

```diff
@@ -0,0 +1,46 @@
+.calendar {
+  display: flex;
+  width: ($day-size * $week-long) + ($cell-gap * ($week-long - 1));
+  padding: 10px;
+  gap: $cell-gap;
+  flex-wrap: wrap;
+  list-style: none;
+
+  @each $day, $day-num in $week-days {
+    &--day-start--#{$day}>:first-child {
+      margin-left: ($day-size + $cell-gap) * $day-num;
+    }
+  }
+
+  @for $day from $min-month-long through $num-days {
+    &--month-length--#{$day}>:NOT(:nth-child(-n+#{$day})) {
+      display: none;
+    }
+  }
+
+  &__cell {
+    box-sizing: border-box;
+    width: $day-size;
+    height: $day-size;
+    cursor: pointer;
+    background-color: $color-day-main;
+    border: $day-border;
+    transition: transform 0.5s;
+
+    &:hover {
+      background-color: $color-day-hover;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through $num-days {
```

Comment:
What does `$i` means?  You should use more meaningful name of the variable

Reply:
changed to $day